src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,53 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int USER_MIN_AGE = 18;
+    private static final int USER_PASSWORD_MIN_LENGTH = 6;
+    private static final int USER_LOGIN_MIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUserInput(user);
+        return storageDao.add(user);
+    }
+
+    private void validateUserInput(User user) {
+        validateUserExist(user);
+        validateUserLogin(user.getLogin());
+        validateUserPassword(user.getPassword());
+        validateUserAge(user.getAge());
+    }
+
+    private void validateUserExist(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserException("User with login '"
+                    + user.getLogin() + "' already exists");
+        }
+    }
+
+    private void validateUserAge(Integer age) {
+        if (age == null || age < USER_MIN_AGE) {
+            throw new InvalidUserException("Users under "
+                    + USER_MIN_AGE + " are not allowed!");
+        }
+    }
```

Comment:
separate the null checks from the other ones, to have a clear message because right now if age is null you are saying it is under the min age, which is misguiding
same for other validation methods